app/Models/Competition.php

```diff
@@ -31,16 +31,19 @@ public function users()
     /**
      * Gets a CSV export of all the users in this competition.
      *
+     * @param \Gladiator\Repositories\UserRepositoryContract $repository
      * @return \League\Csv\ $csv
      */
-    public function getCSVExport()
+    public function getCSVExport($repository)
```

Comment:
@deadlybutter this is where I was also mentioning instead of doing this in the Model, maybe we create a `ContestService` which has a bunch of functionality for dealing with contests, competitions, waitingrooms that don't feel right in a Model (which usually deals with a single item) and to help abstract crazy logic in controllers that might end up getting repeated. Could be a universal service that deals with this stuff; doesn't need to be a separate service each for contests, competitions, waitingrooms...